plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -62,14 +68,27 @@
 
   // Preference store (should be configuration scoped) from which we get UUID, opt-in status, etc.
   private IEclipsePreferences preferences;
-
-  private OptInDialogCreator optInDialogCreator;
+  private Display display;
+
+  private ConcurrentLinkedQueue<PingEvent> pingEventQueue;
```

Comment:
Using an interface is usually nice, but on second thought, I think it's better to explicitly say here that the queue required should be a thread-safe one, to prevent getting a plain `Queue`. I tried to see if there is any other interface such as `ConcurrentQueue`, but there doesn't seem to.